server/src/main/java/io/crate/types/DataTypes.java

```diff
@@ -610,6 +610,19 @@ public static boolean isCompatibleType(List<DataType<?>> left, List<DataType<?>>
         return true;
     }
 
+    public static DataType<?> merge(DataType<?> leftType, DataType<?> rightType) {
+        DataType<?> type;
+        if (leftType.id() == ObjectType.ID) {
+            type = ObjectType.merge((ObjectType) leftType, (ObjectType) rightType);
```

Comment:
This fails if the right type isn't an object

Reply:
Thanks for pointing out, I just blindly took it out of UnionSelect.